File: DotNet/Node.Core/Biz/Interfaces/Download/IPreProcess.cs
using System;

using Node.Core.Biz.Manageable.Parameters;
using Node.Core.Document;

namespace Node.Core.Biz.Interfaces.Download
{
    /// <summary>
    /// The Interface for Download PreProcess
    /// </summary>
    public interface IPreProcess
    {
        /// <summary>
        /// PreProcess of Download.
        /// </summary>
        /// <param name="token">The Token which is result of Authentication Operation.</param>
        /// <param name="transID">The transaction id is result of Solicict/Submiet Operation.</param>
        /// <param name="dataFlow">Specifies which dataflow are to be used. </param>
        /// <param name="docs">The reserved container for the download requesed documents.</param>
        /// <param name="param">Operation related parameters</param>
        /// <returns>Array of NodeDocument</returns>
        void Execute(string token, string transID, string dataFlow, NodeDocument[] docs, PreParam param);
    }
}

File: EN Node for Java environments/Node/src/Node/Biz/Interfaces/Download/IPreProcess.java
package Node.Biz.Interfaces.Download;

import java.rmi.RemoteException;

import Node.Biz.Custom.PreParam;
import Node.WebServices.Document.ClsNodeDocument;
/**
 * <p>This class create Authenticate Pre Process Interface.</p>
 * <p>Company: enfoTech & Consulting, Inc.</p>
 * @author enfoTech
 * @version 2.0
 */


public interface IPreProcess {
	/**
	 * Execute
	 * @param token The authentication token
	 * @param transID The transaction ID
	 * @param dataFlow The data flow name
	 * @param docs The download file object array
	 * @param param The PreParam object 
	 * @throws RemoteException
	 * @return PreParam object 
	 */
  public PreParam Execute (String token, String transID, String dataFlow, ClsNodeDocument[] docs, PreParam param) throws RemoteException;
}
